topk-py/src/client/async/collections.rs

```diff
@@ -0,0 +1,95 @@
+use crate::data::collection::Collection;
+use crate::error::RustError;
+use crate::schema::field_spec::FieldSpec;
+use pyo3::prelude::*;
+use pyo3_async_runtimes::tokio::future_into_py;
+use std::{collections::HashMap, sync::Arc};
+use topk_rs::proto::v1::control::FieldSpec as FieldSpecPb;
+
+#[pyclass]
+pub struct AsyncCollectionsClient {
+    client: Arc<topk_rs::Client>,
+}
+
+impl AsyncCollectionsClient {
+    pub fn new(client: Arc<topk_rs::Client>) -> Self {
+        Self { client }
+    }
+}
+
+#[pymethods]
+impl AsyncCollectionsClient {
+    pub fn get(&self, py: Python<'_>, collection_name: String) -> PyResult<PyObject> {
+        let client = self.client.clone();
+
+        future_into_py(py, async move {
+            let collection = client
+                .collections()
+                .get(&collection_name)
+                .await
+                .map_err(RustError)?;
+
+            let collection: Collection = collection.into();
+
+            Ok(collection)
+        })
+        .map(|result| result.into())
+    }
+
+    pub fn list(&self, py: Python<'_>) -> PyResult<PyObject> {
+        let client = self.client.clone();
+
+        future_into_py(py, async move {
+            let collections = client.collections().list().await.map_err(RustError)?;
+
+            Ok(collections
+                .into_iter()
+                .map(|i| i.into())
+                .collect::<Vec<Collection>>())
+        })
+        .map(|result| result.into())
+    }
+
+    pub fn create(
+        &self,
+        py: Python<'_>,
+        collection_name: String,
+        schema: HashMap<String, FieldSpec>,
+    ) -> PyResult<PyObject> {
+        let client = self.client.clone();
+
+        future_into_py(py, async move {
+            let collection = client
+                .collections()
+                .create(
+                    &collection_name,
+                    schema
```

Comment:
nit: having `Schema` object here would be nicer so we don't have to duplicate the pb conversion

Reply:
updated in https://github.com/topk-io/topk/pull/233/commits/71d9a14ae1ddea5990f22d0a4a32745e511bdd33